code-challenges.test.js

```diff
@@ -15,35 +15,125 @@
+    it('takes in an array, removes the first item from the array and shuffles the remaining content', () => {
+      expect(remFirstShuff(colors1)).toEqual(expect.arrayContaining(colors1));
+      expect(remFirstShuff(colors2)).toEqual(expect.arrayContaining(colors2));
+    });
+  });
 
 
-// --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
+// // b) Create the function that makes the test pass.
 
-// a) Create a test with expect statements for each of the variables provided.
 
-const nums1 = [3, 56, 90, -8, 0, 23, 6]
-// Expected output: [-8, 90]
-const nums2 = [109, 5, 9, 67, 8, 24]
-// Expected output: [5, 109]
+// // create a function called "remFirstShuff" 
+// // input param of "array"
+// // use shift() method to remove first element of array
+// // create a for loop to iterate through the array
+// // include Math.random() method to randomize the array
+// // return the new randomized array
 
+const remFirstShuff = (array) => {
 
-// b) Create the function that makes the test pass.
+    array.shift()  
+
+    for (let i = array.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i + 1));
+        [array[i], array[j]] = [array[j], array[i]];
+    }
+
+    return array
+
+}
+
+
+
+
+// // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
+
+// // a) Create a test with expect statements for each of the variables provided.
+
+// const nums1 = [3, 56, 90, -8, 0, 23, 6]
+// // Expected output: [-8, 90]
+// const nums2 = [109, 5, 9, 67, 8, 24]
+// // Expected output: [5, 109]
+
+describe('minMax', () => {
+    const nums1 = [3, 56, 90, -8, 0, 23, 6]
+    const nums2 = [109, 5, 9, 67, 8, 24]
+    it('takes an array of numbers and returns an array of the minimum and maximum numbers in that order', () => {
+        expect(minMax(nums1)).toEqual([-8, 90]);
+        expect(minMax(nums2)).toEqual([5, 109]);
+    });
+  });
+
+
+// // b) Create the function that makes the test pass.
 
+// // create a function called "minMax"
+// // create param called "array"
+// // create a variable called "max" and "min"
+// // use reduce() method to obtain only 1 value
+// // create a function and use the Math.max/Math.min methods to obtain the max number and min number
+// // return the min and max number
 
-// --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
+const minMax = (array) => {
+    
+    var max = array.reduce(function(a, b) {
+        return Math.max(a, b) 
+    })
 
-// a) Create a test with an expect statement using the variables provided.
+    var min = array.reduce(function(a, b) {
+        return Math.min(a, b) 
+    })
+
+    return [min, max] 
+
+}
```

Comment:
Looking up what Math.min / .max do would really help with the refactoring process.